### ISSUE TITLE
ElectrumX Onion host support

### ISSUE THREAD

**Issue by java4tech** [sparrowwallet/sparrow#9]
Do you have any plans to support tor?

**Comment by craigraw** [comment]
Yes - working on that next.

**Comment by craigraw** [comment]
It's already possible to connect Sparrow to a Tor hidden service (.onion address) if you are running tor locally - just set the proxy details in the server preferences appropriately e.g. 127.0.0.1:9050. 

By supporting tor, do you mean bundling the tor binaries with Sparrow so that it does not need a locally running tor instance?

**Comment by java4tech** [comment]
I have tried to test it on Windows and got error on port 9050, I was getting connection refused but 9150 worked but still got the following error.
```
Could not connect:

Unsupported or unrecognized SSL message
```

**Comment by java4tech** [comment]
I was thinking about a use case where we can connect to any public node only for Sending Transaction, This can be used in a situation where user want's to use there server only for transaction update and don't want to reveal his node's IP while sending payment.
Some suggestions: 
1. You can use [Netlayer](https://github.com/JesusMcCloud/netlayer) project to support native tor. It is used by [bisq](https://bisq.network/) project.
2. It'll be nice to have an option to disconnect node, user might want to do certain tasks while offline.

**Comment by craigraw** [comment]
Implemented built in Tor support and released in v0.9.4 - closing this off.

The idea to have a connection to a separate node for broadcasting is an interesting one - perhaps something for a future release.
